IoTCoreDefaultApp/IoTCoreDefaultApp/Views/Settings.xaml.cs

```diff
@@ -254,5 +388,548 @@ private void RefreshButton_Click(object sender, RoutedEventArgs e)
+            {
+                formatString = BluetoothDeviceInformationDisplay.GetResourceString("BluetoothUnpairingFailureFormat");
+                confirmationMessage = string.Format(formatString, unpairingResult.Status.ToString(), deviceInfoDisp.Name, deviceInfoDisp.Id);
+            }
+            // Display the result of the pairing attempt
+            DisplayMessagePanel(confirmationMessage, MessageType.InformationalMessage);
+
+            // Clear any devices in the listand stop and restart the watcher to ensure state is reflected in list
+            bluetoothDeviceObservableCollection.Clear();
+            StopWatcher();
+            StartWatcher();
+
+            // Re-enable the unpair button
+            unpairButton.IsEnabled = true;
+        }
+
+        /// <summary>
+        /// User has entered a PIN and pressed <Return> in the PIN entry flyout
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void PinEntryTextBox_KeyDown(object sender, Windows.UI.Xaml.Input.KeyRoutedEventArgs e)
+        {
+
+            if (e.Key == Windows.System.VirtualKey.Enter)
+            {
+                //  Close the flyout and save the PIN the user entered
+                TextBox bluetoothPINTextBox = sender as TextBox;
+                string pairingPIN = bluetoothPINTextBox.Text;
+                if (pairingPIN != "")
+                {
+                    // Hide the flyout
+                    inProgressPairButton.Flyout.Hide();
+                    inProgressPairButton.Flyout = null;
+                    // Use the PIN to accept the pairing
+                    AcceptPairingWithPIN(pairingPIN);
+                }
+            }
+        }
+
+        private void AcceptPairingWithPIN(string PIN)
+        {
+            if (pairingRequestedHandlerArgs != null)
+            {
+                pairingRequestedHandlerArgs.Accept(PIN);
+                pairingRequestedHandlerArgs = null;
+            }
+            // Complete the deferral here
+            CompleteDeferral();
+        }
+
+        /// <summary>
+        /// Call when selection changes on the list of discovered Bluetooth devices
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void ResultsListView_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+
+        }
+
+        /// <summary>
+        /// Get the set of acceptable ceremonies from the check boxes
+        /// </summary>
+        /// <returns></returns>
+        private DevicePairingKinds GetSelectedCeremonies()
+        {
+            DevicePairingKinds ceremonySelection = DevicePairingKinds.ConfirmOnly | DevicePairingKinds.DisplayPin | DevicePairingKinds.ProvidePin | DevicePairingKinds.ConfirmPinMatch;
+            //if (confirmOnlyOption.IsChecked.HasValue && (bool)confirmOnlyOption.IsChecked) ceremonySelection |= DevicePairingKinds.ConfirmOnly;
```

Comment:
How should I reply? On the GitHub web site?

The reason is that people couldn't make up their minds whether to allow ceremony selection via toggle buttons or just use the OR to get all of them so I left the code.

Sent from my Windows Phone

-----Original Message-----
From: "Brian Fjeldstad" <notifications@github.com>
Sent: ‎10/‎26/‎2015 4:05 PM
To: "ms-iot/samples" <samples@noreply.github.com>
Cc: "Kara Ottewell" <karaottewell@outlook.com>
Subject: Re: [samples] Changes to add the Bluetooth tab to Settings so that BT Inbound and Outbound Pairing can be done (#130)

In IoTCoreDefaultApp/IoTCoreDefaultApp/Views/Settings.xaml.cs:
> +        /// </summary>
> +        /// <param name="sender"></param>
> +        /// <param name="e"></param>
> +        private void ResultsListView_SelectionChanged(object sender, SelectionChangedEventArgs e)
> +        {
> +
> +        }
> +
> +        /// <summary>
> +        /// Get the set of acceptable ceremonies from the check boxes
> +        /// </summary>
> +        /// <returns></returns>
> +        private DevicePairingKinds GetSelectedCeremonies()
> +        {
> +            DevicePairingKinds ceremonySelection = DevicePairingKinds.ConfirmOnly | DevicePairingKinds.DisplayPin | DevicePairingKinds.ProvidePin | DevicePairingKinds.ConfirmPinMatch;
> +            //if (confirmOnlyOption.IsChecked.HasValue && (bool)confirmOnlyOption.IsChecked) ceremonySelection |= DevicePairingKinds.ConfirmOnly;
why all the commented code?
—
Reply to this email directly or view it on GitHub.